api/envoy/config/core/v3/base.proto

```diff
@@ -320,12 +320,37 @@ message HeaderValueOption {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.core.HeaderValueOption";
 
+  // Describes the supported actions types to be used in :ref:`append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>`
+  enum HeaderAppendAction {
+    // This action will append the specified value to the existing values if the header
+    // already exists. If the header doesn't exist then this will add the header with
+    // specified key and value.
+    APPEND_IF_EXISTS = 0;
+
+    // This action will add the header if it doesn't already exist. If the header
+    // already exists then this will be a no-op.
+    ADD_IF_ABSENT = 1;
+
+    // This action will overwrite the specified value by discarding any existing values if
+    // the header already exists. If the header doesn't exist then this will add the header
+    // with specified key and value.
+    OVERWRITE_IF_EXISTS = 2;
+  }
```

Comment:
The name `APPEND_IF_EXISTS` does not seem to be appropriate. It always adds header key/value to the headers although the header key is not exist.
This action behavior is not what it literally implies.
`OVERWRITE_IF_EXISTS` has a similar problem.

Reply:
Any suggestions on what could be a more appropriate name? Maybe something like "APPEND_IF_EXISTS_OR_ADD"?

Reply:
In my personal opinion, I think may be simple `APPEND` is enough. Append new header field, whether it already exists or not.
Similarly, `OVERWRITE` is enough.